### DISCUSSION TITLE
Background color doesnt work(react js)

### DISCUSSION THREAD

**Discussion by smoke-flame** [tailwindlabs/tailwindcss#10490] [Help] [answered]
I am using tailwindcss ^3.2.4. When I pass bg color as a props in component and use it on element it doesnt work. If I write 'bg-red-600' it works, but  if I use color from props in react it wont work. Its pretty weird because if i write bg-red-600 and then replace it with  `bg-${color}-600` it will be red with hot reload. But after page reloading it becomes transparent again.

![image](https://user-images.githubusercontent.com/75222498/216784470-c3666d54-ef3d-44ca-92de-a481035552df.png)
![image](https://user-images.githubusercontent.com/75222498/216784362-e797d6f4-cd9d-4e47-a6dd-6f5539368f0c.png)

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):
> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as complete unbroken strings in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS

**Reply by smoke-flame** [reply]
> As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):
> 
> > The most important implication of how Tailwind extracts class names is that it will only find classes that exist as complete unbroken strings in your source files.
> > If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS

Got it. Thank you. Its not really convenient to write config for each color, but if it works so, then ok. Hope its going to be possible to deal with that in more pure way.
